### PR TITLE
Allow parameters to be hidden from help text

### PR DESCRIPTION
**Pull request by natemcmaster**
cc @NTaylorMullen so we can parse dispatcher version using the command line app but not put it the help text.

### CODE REVIEW

File: test/Microsoft.Extensions.CommandLineUtils.Tests/CommandLineApplicationTests.cs [hunk 1]
```diff
@@ -501,5 +501,45 @@ public void NestedInheritedOptions()
...
+                Name = "ninja-app",
+                Description = "You can't see it until it is too late"
+            };
+
+            app.Command("star", c =>
+            {
+                c.Option("--points <p>", "How many", CommandOptionType.MultipleValue);
+                c.Hidden = true;
+            });
+            app.Option("--no-kill", "Be a nice ninja", CommandOptionType.NoValue, o => { o.Hidden = true; });
```
Comment (pranavkm) [inline test/Microsoft.Extensions.CommandLineUtils.Tests/CommandLineApplicationTests.cs]: Umm, maybe a different name? Not super sure if policheck likes this.

File: src/Microsoft.Extensions.CommandLineUtils/CommandLine/CommandOption.cs [hunk 2]
```diff
@@ -59,7 +59,7 @@ public CommandOption(string template, CommandOptionType optionType)
         public string Description { get; set; }
         public List<string> Values { get; private set; }
         public CommandOptionType OptionType { get; private set; }
-
+        public bool Hidden { get; set; }
```
Comment (pranavkm) [inline src/Microsoft.Extensions.CommandLineUtils/CommandLine/CommandOption.cs]: `ShowInHelpText { get; set;} = true`?

**Comment by pranavkm** [comment]
Changes look ok. Maybe rename some of the test before checking it in.

:shipit:

File: test/Microsoft.Extensions.CommandLineUtils.Tests/CommandLineApplicationTests.cs (hunk 1 above)
Comment (natemcmaster) [inline test/Microsoft.Extensions.CommandLineUtils.Tests/CommandLineApplicationTests.cs]: anyway simple to test polycheck locally?

Comment (pranavkm) [inline test/Microsoft.Extensions.CommandLineUtils.Tests/CommandLineApplicationTests.cs]: I think @dougbu sent an email a while ago about looking up their db.

File: src/Microsoft.Extensions.CommandLineUtils/CommandLine/CommandOption.cs (hunk 2 above)
Comment (natemcmaster) [inline src/Microsoft.Extensions.CommandLineUtils/CommandLine/CommandOption.cs]: Not opposed, just wondering...is there something else `Hidden` could mean?

Comment (pranavkm) [inline src/Microsoft.Extensions.CommandLineUtils/CommandLine/CommandOption.cs]: It's not super obvious what is being hidden :)

File: test/Microsoft.Extensions.CommandLineUtils.Tests/CommandLineApplicationTests.cs (hunk 1 above)
Comment (natemcmaster) [inline test/Microsoft.Extensions.CommandLineUtils.Tests/CommandLineApplicationTests.cs]: The terms pass my search on the db, but I can change it if there's any chance it'll cause trouble.

File: src/Microsoft.Extensions.CommandLineUtils/CommandLine/CommandOption.cs (hunk 2 above)
Comment (natemcmaster) [inline src/Microsoft.Extensions.CommandLineUtils/CommandLine/CommandOption.cs]: Ok. will update.

File: test/Microsoft.Extensions.CommandLineUtils.Tests/CommandLineApplicationTests.cs (hunk 1 above)
Comment (pranavkm) [inline test/Microsoft.Extensions.CommandLineUtils.Tests/CommandLineApplicationTests.cs]: Cool. No worries in that case :joy_cat:

**Pull request merged**